assets/js/modules/adsense/components/module/ModuleTopEarningPagesWidget.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * ModuleTopEarningPagesWidget module
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { numFmt } from '../../../../util';
+import ReportTable from '../../../../components/ReportTable';
+import Link from '../../../../components/Link';
+import PreviewTable from '../../../../components/PreviewTable';
+import TableOverflowContainer from '../../../../components/TableOverflowContainer';
+import AdSenseLinkCTA from '../../../analytics/components/common/AdSenseLinkCTA';
+import { STORE_NAME, DATE_RANGE_OFFSET } from '../../datastore/constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+import { isZeroReport } from '../../../analytics/util/is-zero-report';
+import { isRestrictedMetricsError } from '../../../analytics/util/error';
+import { generateDateRangeArgs } from '../../../analytics/util/report-date-range-args';
+import { getCurrentDateRangeDayCount } from '../../../../util/date-range';
+import ModuleHeader from '../common/ModuleHeader';
+const { useSelect } = Data;
+
+function ModuleTopEarningPagesWidget( { Widget, WidgetReportZero, WidgetReportError } ) {
```

Comment:
@johnPhillips let's split this component into smaller ones similarly to how Analytics module components are structured (see [here](https://github.com/google/site-kit-wp/tree/develop/assets/js/modules/analytics/components/module)).